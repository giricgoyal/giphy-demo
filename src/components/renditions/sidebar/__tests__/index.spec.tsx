import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Sidebar, { Props } from '../index'

describe('<Sidebar />', () => {
    let container, getAllByTestId
    const onSidebarItemClickHandler = jest.fn()

    const defaultProps: Props = {
        items: ['captain_america', 'black_widow', 'thor'],
        selectedItem: 'thor',
        onSidebarItemClick: onSidebarItemClickHandler,
    }

    beforeEach(() => {
        ;({ container, getAllByTestId } = render(<Sidebar {...defaultProps} />))
    })

    test('should match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('onSidebarItemClick', () => {
        describe('when an item is clicked', () => {
            beforeEach(() => {
                fireEvent.click(getAllByTestId('sidebar-item')[1])
            })

            test('should call the onSidebarItemClick', () => {
                expect(onSidebarItemClickHandler).toHaveBeenCalledWith('black_widow')
            })
        })
    })
})
