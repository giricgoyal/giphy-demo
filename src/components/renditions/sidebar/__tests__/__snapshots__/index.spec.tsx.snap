// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should match snapshot 1`] = `
<div>
  <div
    class="renditions__sidebar"
  >
    <div
      class="renditions__sidebar__item "
      data-testid="sidebar-item"
    >
      Captain America
    </div>
    <div
      class="renditions__sidebar__item "
      data-testid="sidebar-item"
    >
      Black Widow
    </div>
    <div
      class="renditions__sidebar__item active"
      data-testid="sidebar-item"
    >
      Thor
    </div>
  </div>
</div>
`;
